extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -89,27 +92,35 @@ public Task StartAsync(CancellationToken cancellationToken)
                 var input = new TriggeredFunctionData() { TriggerValue = ea };
                 FunctionResult result = await this.executor.TryExecuteAsync(input, cancellationToken).ConfigureAwait(false);
 
-                if (result.Succeeded)
-                {
-                    this.rabbitMQModel.BasicAck(ea.DeliveryTag, false);
-                }
-                else
+                if (!result.Succeeded)
                 {
-                    if (ea.BasicProperties.Headers == null || !ea.BasicProperties.Headers.ContainsKey(Constants.RequeueCount))
-                    {
-                        this.CreateHeadersAndRepublish(ea);
-                    }
-                    else
+                    ea.BasicProperties.Headers ??= new Dictionary<string, object>();
+                    ea.BasicProperties.Headers.TryGetValue(RequeueCountHeaderName, out object headerValue);
+                    int requeueCount = Convert.ToInt32(headerValue, CultureInfo.InvariantCulture) + 1;
+
+                    if (requeueCount >= 5)
```

Comment:
Behavior is changed here. Same message will trigger the user function 5 times instead of 6 times. There was code duplication between `CreateHeadersAndRepublish` and `RepublishMessages` requiring the same fix to be added to both methods, hence trying to simplify the logic and removing redundancy.